Tests/StreamChatTests/Database/DatabaseContainer_Tests.swift

```diff
@@ -193,6 +119,45 @@ final class DatabaseContainer_Tests: XCTestCase {
             }
         }
     }
+    
+    func test_removingAllData_whileAnotherWrite() throws {
+        let container = DatabaseContainer(kind: .inMemory)
+        try writeALotOfData(to: container)
+        
+        // Schedule saving just before removing it all
+        container.write { session in
+            try session.saveChannel(payload: self.dummyPayload(with: .unique), query: nil, cache: nil)
+        }
+        
+        let expectation = XCTestExpectation(description: "Remove")
+        container.removeAllData { error in
+            XCTAssertNil(error)
+            expectation.fulfill()
+        }
+        
+        // Save just after triggering remove all
+        container.write { session in
+            try session.saveChannel(payload: self.dummyPayload(with: .unique), query: nil, cache: nil)
+        }
```

Comment:
Without `canWriteData` this would cause a test failure in this test (but sometimes, because this depends on the timing. I got test failures when I repeated the test 100 times)